src/main/java/gregtech/api/recipes/ingredients/GTRecipeFluidInput.java

```diff
@@ -67,30 +68,30 @@ public boolean acceptsFluid(@Nullable FluidStack input) {
         return false;
     }
 
+    @Override
+    public int hashCode() {
+        if (nbtMatcher == null) {
+            //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
+            return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, inputStack.tag);
+        }
+        return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, 0);
+    }
+
     @Override
     public boolean equals(Object obj) {
         if (this == obj) return true;
         if (!(obj instanceof GTRecipeFluidInput)) {
             return false;
         }
         GTRecipeFluidInput other = (GTRecipeFluidInput) obj;
-        if (this.amount != other.amount) return false;
-        if (this.isConsumable != other.isConsumable) return false;
-        if (this.nbtMatcher != null && !this.nbtMatcher.equals(other.nbtMatcher)) return false;
-        if (this.nbtCondition != null && !this.nbtCondition.equals(other.nbtCondition)) return false;
-        if (inputStack.getFluid().getName().equals(other.inputStack.getFluid().getName())) {
-            return FluidStack.areFluidStackTagsEqual(this.inputStack, other.inputStack);
-        }
-        return false;
-    }
 
-    @Override
-    public int hashCode() {
-        if (nbtMatcher == null) {
-            //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
-            return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, inputStack.tag);
-        }
-        return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, 0);
+        if (this.amount != other.amount || this.isConsumable != other.isConsumable) return false;
+        if (!Objects.equals(this.nbtMatcher, other.nbtMatcher)) return false;
+        if (!Objects.equals(this.nbtCondition, other.nbtCondition)) return false;
+
+        //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
+        return inputStack.getFluid().getName().equals(other.inputStack.getFluid().getName()) &&
+                FluidStack.areFluidStackTagsEqual(this.inputStack, other.inputStack);
```

Comment:
Would `FluidStack#isFluidEqual` work in place of this code?

Reply:
To me checking fluid's name instead of refchecking looks like a deliberate choice. I'm not aware of the situations where a simple refcheck breaks, though. I just trusted the comment.